src/google-apis/auth-client.ts

```diff
@@ -110,7 +110,9 @@ export class RequestHandler extends Service {
     this._config = config;
     this._logger = logger;
 
-    if (tryAuthenticate) {
+    if (!this._config.getShouldReportErrorsToAPI()) {
```

Comment:
I believe you need to add an extra parameter controlling your feature (e.g. something like `authenticateIfErrorReportingIsOff` which by default should be `on`). The `getShouldReportErrorsToAPI` is existing feature and if for some reason this change contains a bug, there is no way to get `getShouldReportErrorsToAPI` working - e.g. we cannot tight the `getShouldReportErrorsToAPI` as the only flag controlling both features, hope it makes sense.

Reply:
I don't think of these as two separate features: _authentication exists only to report errors to the API_. If we are never reporting errors, why would we authenticate? It's not expected and that's why I believe `getShouldReportErrorsToAPI` should be used for both.

If there were a separate use-case aside from error reporting or other public functions in this client, then sure, they could be detached. But right now, if I were to make another config option, it unnecessarily complicates the configuration; it can only lead to cases where people get unexpected behavior by having the settings out of sync.

Reply:
To be honest, I am not very familiar with this code since I only recently was assigned to this repo. I agree with your comment and I think that adding redundant config is a bad practice. Thats said, this client library is used by many developers and any bugs related to this change might have an impact on existing users - thats a reason I asked for extra parameter. We can definitely make it right, so I would really appreciate if you can confirm if you ran this change E2E. In addition, would you be able to add system test for this functionality?

Reply:
Ok, yes I'll have this run in production for us for a few days and then also look into a system test. I had some trouble running those:
```
The project id (projectId) was not set in the env
The api key (apiKey) was not set as an env variable
The project number (projectNumber) was not set in the env
The key filename (keyFilename) was not set in the env
Skipping error-reporting system tests
``` 
Which is why I didn't write one initially, so I might be missing information on getting those to work but will look again.

Reply:
Ok, was able to get them going and wrote a new system-test assertion that fails with the old code and passes with the new.

Reply:
OK, thanks